src/main/webapp/3Dscript.js

```diff
@@ -16,10 +16,54 @@ const poseTransform = {
   /* Unit: 4 meters*/ translateZ: 0,
   /* Unit: degrees*/ rotate: 0,
   /* Scalar*/ scale: 1};
+
+const poseStart = {start: 0};
+const poseEnd = {end: 10000};
+const raycaster = new THREE.Raycaster();
+const mouse = new THREE.Vector2();
+
 const apiKey = 'AIzaSyDCgKca9sLuoQ9xQDfHUvZf1_KAv06SoTU';
 
+/* Matrices used to scale and unscale pose cylinders */
+const scaleCylinder = 2;
+const scaleMatrix = new THREE.Matrix4().makeScale(
+    scaleCylinder, scaleCylinder, scaleCylinder );
+const scaleInverseMatrix = new THREE.Matrix4().makeScale(
+    1/scaleCylinder, 1/scaleCylinder, 1/scaleCylinder );
+
+let selectedIndex= -1;
+
+/* For hiding trajectory. */
+const scaleHide = 400;
+const zeroMatrix = new THREE.Matrix4().makeScale(
+    1/scaleHide, 1/scaleHide, 1/scaleHide );
+const nonZeroMatrix = new THREE.Matrix4().makeScale(
+    scaleHide, scaleHide, scaleHide);
+
+let oldStart=0;
+let oldEnd;
+
+const instanceMatrix = new THREE.Matrix4();
+const matrix = new THREE.Matrix4();
+
+/*  Datapoint time interface on gui. */
+let time;
+/*  Datapoint lat interface on gui. */
+let lat;
+/*  Datapoint long interface on gui. */
+let long;
```

Comment:
Better to use "lng" instead "long", since long is an ambiguity name with data type.